crates/matrix-sdk/src/authentication/oidc/tests.rs

```diff
@@ -452,3 +456,48 @@ async fn test_insecure_clients() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[async_test]
+async fn test_management_url_cache() {
+    let server = MatrixMockServer::new().await;
+    let client = server.client_builder().unlogged().build().await;
+    let backend =
+        Arc::new(MockImpl::new().mark_insecure().account_management_uri(server.server().uri()));
+    let oidc = Oidc { client: client.clone(), backend: backend.clone() };
+
+    let tokens = OidcSessionTokens {
+        access_token: "4cc3ss".to_owned(),
+        refresh_token: Some("r3fr3sh".to_owned()),
+        latest_id_token: None,
+    };
+
+    let session = mock_session(tokens.clone());
+    oidc.restore_session(session.clone())
+        .await
+        .expect("We should be able to restore an OIDC session");
+
+    // The cache should not contain the entry.
+    assert!(!client
+        .inner
+        .caches
+        .account_management_url
+        .lock()
+        .await
+        .contains(&serde_json::to_string::<Option<AccountManagementAction>>(&None).unwrap()));
```

Comment:
light suggestion: consider a local variable for the atrocity in parenthesis? (Or could we use `Some()` variant of `AccountManagementAction`, so that we don't have to precise the generic type used in `to_string()`?)

Reply:
https://github.com/matrix-org/matrix-rust-sdk/pull/4663/commits/768ec36376896384407fcfa42d968d5344402075